tools/src/lib.rs

```diff
@@ -729,19 +730,42 @@ impl std::str::FromStr for MeshFormat {
     }
 }
 
-/// Write a mesh to stdout in the given format.
-pub fn write_mesh(mesh: &Mesh, format: MeshFormat) -> Result<()> {
-    match format {
-        MeshFormat::MeditAscii => println!("{}", mesh.display_medit_ascii()),
-        MeshFormat::MeditBinary => {
+/// Read a mesh either from stdin or from a file.
+pub fn read_mesh(filename: Option<&String>) -> Result<Mesh> {
+    Ok(match filename.cloned().as_deref() {
+        None | Some("-") => {
+            let stdin = io::stdin();
+            let stdin = stdin.lock();
+            let stdin = io::BufReader::new(stdin);
+            Mesh::from_reader(stdin).context("failed to read mesh from stdin")?
+        }
+        Some(filename) => Mesh::from_file(filename).context("failed to read mesh from file")?,
+    })
+}
+
+/// Write a mesh, either to stdout or to a file, in the given format.
+pub fn write_mesh(mesh: &Mesh, format: MeshFormat, filename: Option<&String>) -> Result<()> {
+    fn write(mut out: impl io::Write, mesh: &Mesh, format: MeshFormat) -> io::Result<()> {
+        match format {
+            MeshFormat::MeditAscii => writeln!(out, "{}", mesh.display_medit_ascii())?,
+            MeshFormat::MeditBinary => mesh.serialize_medit_binary(out)?,
+        }
+        Ok(())
+    }
+
+    match filename.cloned().as_deref() {
+        None | Some("-") => {
             let stdout = io::stdout();
             let stdout = stdout.lock();
-            let stdout = io::BufWriter::new(stdout);
-            mesh.serialize_medit_binary(stdout)
-                .context("failed to write mesh")?;
+            let writer = io::BufWriter::new(stdout);
+            write(writer, mesh, format).context("failed to write mesh to stdout")
```

Comment:
Looks weird to duplicate call to `write`.

Reply:
Extracted to stdout/filename/"-" logic into a helper function here

https://github.com/LIHPC-Computational-Geometry/coupe/pull/230/files#diff-fa5efd80df43c0b4a9776ca017642d9cb73ca5cd3b69b542577f7ff9bb3fbbd4R748